### PR TITLE
feat(devtools): add new tab `client/routes/doctor` to show overview of rsdoctor

### PR DESCRIPTION
**Pull request by Asuka109**
## Summary

image

- Parse binary manifest of rsdoctor: packages/devtools/kit/src/rsdoctor.ts
- Add new builtin tab to show overview of rsdoctor: packages/devtools/client/src/entries/client/routes/doctor/page.tsx
- Some reusable components.

This is still a WIP and will be merged into the development branch for further iterations.

## Checklist

- [ ] I have added changeset via `pnpm run change`.
- [ ] I have updated the documentation.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: packages/devtools/client/src/entries/client/routes/doctor/page.tsx [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+        gap="4"
+        mb="8"
+        justify="center"
+        className={styles.container}
+      >
+        <IndicateCard className={styles.primaryCard}>
+          <IndicateCard.Column>
+            <Text>Rsdoctor</Text>
+            <Flex gap="2">
+              <button type="button">v1.2.0</button>
```
Comment (chenjiahan) [inline packages/devtools/client/src/entries/client/routes/doctor/page.tsx:51]: which version is this?

Comment (easy1090) [reply packages/devtools/client/src/entries/client/routes/doctor/page.tsx:51]: May this position can show version of webpack or rspack, this value is more important for users.

File: packages/devtools/kit/src/rsdoctor.ts [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+  const { Url } = await import('@rsdoctor/utils/common');
+  if (Url.isUrl(url)) {
+    const resp = await fetch(url);
+    return resp.text();
+  }
+  // json string
+  return url;
+}
+
+export async function parseManifest(
```
Comment (easy1090) [inline packages/devtools/kit/src/rsdoctor.ts:13]: This function may can migarate to rsdoctor repo, later. Would you want to do this ?

File: packages/devtools/client/src/entries/client/routes/doctor/page.tsx (hunk 1 above)
Comment (easy1090) [inline packages/devtools/client/src/entries/client/routes/doctor/page.tsx:51]: May this position can show version of webpack or rspack, this value is more important for users.

**Pull request merged**